src/Dialog.tsx

```diff
@@ -198,33 +199,36 @@ export default class Dialog extends React.Component<IDialogPropTypes, any> {
 
     const style = assign({}, props.style, dest);
     const transitionName = this.getTransitionName();
-    const dialogElement = (
-      <LazyRenderBox
-        key="dialog-element"
-        role="document"
-        ref={this.saveRef('dialog')}
-        style={style}
-        className={`${prefixCls} ${props.className || ''}`}
-        visible={props.visible}
-      >
-        <div className={`${prefixCls}-content`}>
-          {closer}
-          {header}
-          <div
-            className={`${prefixCls}-body`}
-            style={props.bodyStyle}
-            ref="body"
-            {...props.bodyProps}
-          >
-            {props.children}
+    let dialogElement;
+    if (props.visible || !props.destroyOnHide) {
```

Comment:
Sorry for the delayed response. But I think [here is](https://github.com/react-component/dialog/blob/6546b60e77d2eae0e61ef51d19ccfb9b5db5221d/src/DialogWrap.tsx#L62-L71) the better place to add this conditional statement.

Reply:
In that case leaving animation won't work because we are using <Animate inside Dialog and if we unmount its parent then leaving animation won't work @yesmeck

Reply:
Hmm, you are right, how about put it to the below `{(props.visible || !props.destroyOnHide) && dialogElement}`

Reply:
make sense will update the pr

Reply:
Thanks, could you write a test?

Reply:
sure

Reply:
@yesmeck done